gdsfactory/boolean.py

```diff
@@ -11,8 +12,8 @@
 
 
 def boolean(
-    A: ComponentOrReference,
-    B: ComponentOrReference,
+    A: ComponentOrReference | kdb.Shape,
+    B: ComponentOrReference | kdb.Shape,
```

Comment:
Why would you pass a `kdb.Shape` to the boolean operation? Pretty sure you mean `kdb.Shapes`? Otherwise you are getting a single polygonor similar. I guess we could support that, but why would you want that?

If you want that fine grained boolean support, I would recommend directly using `gf.kdb` instead of this boolean support.

Reply:
Passing a `kdb.Shape` to boolean operation occurs in this scenario:
```python
import gdsfactory as gf
c = gf.Component()
rec = gf.components.rectangle(size=(100, 10), layer=(1, 0))
points = [
    (0, 0),
    (70, 0),
    (100, 30),
]
polygon = gf.Component().add_polygon(points=points, layer=(1, 0))
c << gf.boolean(rec, polygon, operation="or", layer=(1, 0))
```
Here I just want to do some boolean operation to the rectangle and polygon I just created, and I think it's a pretty intuitive operation.
However `gf.Component().add_polygon()` method returns a `kdb.Shape` rather than `Component`.
And I also noticed that the code I showed worked correctly before #3013, but now there is a error.

Honestly I also don't think accepting `kdb.Shape` as a parameter is a good solution. Why don't we wrap the return value from `add_polygon` as a `Component`.

Do you have better options to solve this problem?

Thank you

Reply:
In this case you should do:

```python
import gdsfactory as gf
c = gf.Component()
rec = gf.components.rectangle(size=(100, 10), layer=(1, 0))
points = [
    (0, 0),
    (70, 0),
    (100, 30),
]
polygon = gf.Component().add_polygon(points=points, layer=(1, 0))
c << gf.boolean(rec, c, operation="or", layer=(1, 0))